openff/evaluator/backends/dask_kubernetes.py

```diff
@@ -225,10 +228,33 @@ def __init__(
             assert isinstance(cluster_kwargs, dict)
             self._cluster_kwargs.update(cluster_kwargs)
 
+    def _get_function_key(self, function, args, kwargs) -> str:
+        """Returns a more useful function key
+
+        Currently all functions passed through Dask are called `wrapped_function`.
+        This returns a key that is either the actual function name, or the
+        Protocol if the function is to execute a protocol.
+        """
+        funckey = funcname(function)
+        if len(args) >= 2:
+            # might be a protocol
+            try:
+                # TODO: might be slow...
```

Comment:
Have you noticed it being slow?

Reply:
Nope, I just thought it might be. If it adds any overhead it's not noticeable amongst everything else going on.